build/Dockerfile.bundle

```diff
@@ -7,7 +7,7 @@ LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
 LABEL operators.operatorframework.io.bundle.package.v1=oadp-operator
 LABEL operators.operatorframework.io.bundle.channels.v1=stable
 LABEL operators.operatorframework.io.bundle.channel.default.v1=stable
-LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.14.0
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.21.0
```

Comment:
label is updated here because RELATED_IMAGE_ discovery were a recent operator-sdk change.
This tell devs to bump their operator-sdk version if they encounter related images generation issues.

We would no longer encounter issue that is supposed to be fixed by https://github.com/operator-framework/operator-sdk/pull/5765 after this change.